tools/ci_build/generate_nuspec_for_native_nuget.py

```diff
@@ -0,0 +1,462 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+import argparse
+import os
+import re
+import sys
+from pathlib import Path
+
+
+# What does the names of our C API tarball/zip files looks like
+# os: win, linux, osx
+def get_package_name(os, cpu_arch):
+    pkg_name = None
+    if os == "win":
+        pkg_name = "onnxruntime-extensions-win-"
+        pkg_name += cpu_arch
+    elif os == "linux":
+        pkg_name = "onnxruntime-extensions-linux-"
+        pkg_name += cpu_arch
+    elif os == "osx":
+        pkg_name = "onnxruntime-extensions-osx-" + cpu_arch
+    return pkg_name
+
+
+# nuget_artifacts_dir: the directory with uncompressed C API tarball/zip files
+# files_list: a list of xml string pieces to append
+# This function has no return value. It updates files_list directly
+def generate_file_list(nuget_artifacts_dir, files_list, include_pdbs):

```

Comment:
nit: it feels like we're greatly overcomplicating things by copying so much from the ORT script which needs to handle a lot more things like building multiple different packages. 

The ORT setup is extremely complicated and hard to understand. The generation script is run from a proj file in the /csharp directory, you have to go find the CI to figure out the args to run that with and how, it writes the nuspec in a build output dir from the ORT build, and packs using custom commands in the proj. None of that is intuitive or simple to debug. 

Would it be simpler to understand/maintain if we just had a script to update a checked in NativeNuget.nuspec like we have currently? We know exactly what files we want the CI to include and those are fixed. The script can update commit id, version, etc. as needed but it would be editing an existing file vs. trying to generate everything dynamically.

If someone wants to test locally I'm starting to think it's easier for them to edit a checked in nuspec file to update the 'src' paths for things they have, and to remove the entries in 'files' for things they don't. Also easy to directly run `nuget pack` once they have done so.